packages/core/src/ui/UIUtil.ts

```
@@ -0,0 +1,114 @@
+              break;
+          }
+        }
+      }
+      if (_meetRootCanvas) {
+        return null;
+      }
+      entity = entity.parent;
+    }
+    return null;
+  }
+}
```

Biome

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)